sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -0,0 +1,290 @@
+import math
+
+import numpy as np
+import dpctl
+import pyopencl
+
+from sklearn_numba_dpex.utils._device import _DeviceParams
+
+from sklearn_numba_dpex.kmeans.kernels import (
+    make_initialize_to_zeros_1dim_float32_kernel,
+    make_initialize_to_zeros_2dim_float32_kernel,
+    make_initialize_to_zeros_3dim_float32_kernel,
+    make_copyto_kernel,
+    make_broadcast_division_kernel,
+    make_center_shift_kernel,
+    make_half_l2_norm_dim0_kernel,
+    make_sum_reduction_1dim_kernel,
+    make_sum_reduction_2dim_kernel,
+    make_sum_reduction_3dim_kernel,
+    make_fused_fixed_window_kernel,
+    make_assignment_fixed_window_kernel,
+)
+
+
+def _check_power_of_2(e):
+    if e != 2 ** (math.log2(e)):
+        raise RuntimeError(f"Expected a power of 2, got {e}")
+    return e
+
+
+class KMeansDriver:
```

Comment:
We could be even more specific:

```suggestion
class LLoydKMeansDriver:
```

Reply:
done